### PR TITLE
adding Dockerfile for building a runnable image, updating README and …

### PR DESCRIPTION
**Pull request by danielwhatmuff**
Hit some annoying config problems with my local Go installation when trying to build the binary. Thought it would be nice to make a little docker build for people to get started more easily.

Have tested the changed locally - so should be good to go.

Ta

### CODE REVIEW

**Comment by danielwhatmuff** [comment]
Also, hit an issue with the Worker and Controller Instance Type parameters. They dont seem to be referenced and so the default is used even if they are specified in the cluster.yaml.

File: multi-node/aws/cmd/kube-aws/command_up.go [hunk 1]
```diff
@@ -128,17 +128,21 @@ func mustReadFile(loc string) []byte {
...
+		AdminKeyFile              string
+		CACertFile                string
 	}{
-		ClusterName:       cfg.ClusterName,
-		APIServerEndpoint: fmt.Sprintf("https://%s", cfg.ExternalDNSName),
-		AdminCertFile:     tlsConfig.AdminCertFile,
-		AdminKeyFile:      tlsConfig.AdminKeyFile,
-		CACertFile:        tlsConfig.CACertFile,
+		ClusterName:                   cfg.ClusterName,
+		ControllerInstanceType:        cfg.ControllerInstanceType,
```
Comment (bcwaldon) [inline multi-node/aws/cmd/kube-aws/command_up.go]: Where are `ControllerInstanceType` and `WorkerInstanceType` actually used in the template below?

**Comment by bcwaldon** [comment]
I'm sort of meh on maintaining a dockerfile & image for a command-line tool. @aaronlevy do you have any strong feelings?

**Comment by danielwhatmuff** [comment]
@bcwaldon it means you can version control your Kubernetes cluster configuration as well as the tool you use to launch it - I've hit problems in the past with things like version differences between aws/vcloud cli tools on laptop to Jenkins. It makes CI for infra more reliable as Jenkins can just run the versioned container to build an environment in aws, then deploy manifests, then run tests.

**Comment by aaronlevy** [comment]
I agree with @bcwaldon. Also, the approach in general I feel a little iffy about. I wouldn't want credentials in docker images the same way I wouldn't want credentials committed in vcs. So now we just get a versioned binary, and a fixed cluster.yaml -- and I wouldn't want to support an image build in this repo just for that.

**Comment by bcwaldon** [comment]
@danielwhatmuff Thank you for the contribution, but I don't feel like this is something we will carry in coreos-kubernetes.

**Pull request closed without merging**